### PR TITLE
FDTN-1043: return an error for invalid base64 encoding

### PR DESCRIPTION
**Pull request by benwaffle**

### CODE REVIEW

File: app/lib/Authorization.scala [hunk 1]
```diff
@@ -108,9 +111,15 @@ class AuthorizationParser @Inject() (
       case prefix :: value :: Nil => {
         prefix.toLowerCase.trim match {
           case Authorization.Prefixes.Basic => {
-            new String(java.util.Base64.getDecoder.decode(value.getBytes(StandardCharsets.US_ASCII))).split(":").toList match {
-              case Nil => Authorization.InvalidApiToken
-              case token :: _ => Authorization.Token(token)
+            Try(new String(java.util.Base64.getDecoder.decode(value.getBytes(StandardCharsets.US_ASCII)))) match {
+              case Failure(ex) =>
+                logger.info("Base64 error", ex)
+                Authorization.OtherError("Failed to decode basic auth credentials")
```
Comment (mbryzek) [inline app/lib/Authorization.scala:117]: Would prefer we have an explicit type here - eg `InvalidBase64Encoding` - these are all internal so no need to be generic

File: app/lib/Authorization.scala [hunk 2]
```diff
@@ -108,9 +111,15 @@ class AuthorizationParser @Inject() (
       case prefix :: value :: Nil => {
         prefix.toLowerCase.trim match {
           case Authorization.Prefixes.Basic => {
-            new String(java.util.Base64.getDecoder.decode(value.getBytes(StandardCharsets.US_ASCII))).split(":").toList match {
-              case Nil => Authorization.InvalidApiToken
-              case token :: _ => Authorization.Token(token)
+            Try(new String(java.util.Base64.getDecoder.decode(value.getBytes(StandardCharsets.US_ASCII)))) match {
```
Comment (mbryzek) [inline app/lib/Authorization.scala:114]: wdyt of using apache commons for this - eg

https://commons.apache.org/proper/commons-net/apidocs/org/apache/commons/net/util/Base64.html#isArrayByteBase64-byte:A-

```
public static boolean [isArrayByteBase64](https://commons.apache.org/proper/commons-net/apidocs/src-html/org/apache/commons/net/util/Base64.html#line.602)(byte[] arrayOctet)
Tests a given byte array to see if it contains only valid characters within the Base64 alphabet. Currently the method treats whitespace as valid.
```

Testing may be less work than decoding (haven't looked at implementation)

**Review by mbryzek — APPROVED** [review]

File: app/lib/Authorization.scala (hunk 2 above)
Comment (benwaffle) [reply app/lib/Authorization.scala:114]: I was actually able to remove the entire dependency just now

Comment (benwaffle) [reply app/lib/Authorization.scala:114]: >Testing may be less work than decoding

Do you think this a CPU bottleneck in proxy?

Comment (mbryzek) [reply app/lib/Authorization.scala:114]: not too worried - but proxy does carry a lot of traffic and this will happen for I think almost every request

Comment (benwaffle) [reply app/lib/Authorization.scala:114]: Proxy is at about [10% CPU usage](https://app.datadoghq.com/metric/explorer?start=1658609176162&end=1658781976162&paused=false#N4IgNghgng9grgFxALlADxQRgDQil3AdwEsATBACwJAoFNiBzCpZTAX11NoDNiA7YgmIw+KUAigAHWihBCAtrQDOtAE7FlIXKtoBHOMoRKUAbVA6lkkSoD63GKvkQWc4opXrNufWo3HkZiB8EIqyPqpQmFogpM4QNkrwqgDGMsggigjqycbeBhGyEABuDMikMMkA1moAdMmScDVwHsAeRcSpyJKqMGhQbAAEAEZQA8AUMEoIbCBsALq49o5wkP6BS-IrEGH5kbMLIFNQYGmgkhAnCAhpMTAMNsmrSh3RYPy0NhLSsolvpK-vGwkchUdJ8BxOMCzTjESyQKCfKQ3N58GTzNhsIA). And this PR doesn't add any extra steps that could make it slower. Testing the validity (of valid headers) might do that.

**Review by mbryzek — APPROVED** [review]

**Pull request merged**